lib/puppet/defaults.rb

```diff
@@ -854,11 +854,23 @@ def self.default_diffargs
         This is distinct from the certificate authority's CRL."
     },
     :certificate_revocation => {
-        :default  => true,
-        :type     => :boolean,
-        :desc     => "Whether certificate revocation should be supported by downloading a
-          Certificate Revocation List (CRL)
-          to all clients.  If enabled, CA chaining will almost definitely not work.",
+        :default  => 'chain',
+        :type     => :certificate_revocation,
+        :desc     => <<EOT
+Whether certificate revocation checking should be enabled, and what level of checking should be performed.
+
+When certificate_revocation is set to 'true' or 'chain', Puppet will download the CA CRL and will perform revocation
+checking against each certificate in the chain. When Puppet is using a single root certificate and has no intermediates
```

Comment:
Agreed this has long been both incomplete and inaccurate. That said I advocate for having the docs/info as accurate as possible for the code they represent at any given time so since we're in here making improvements we might as well be honest about the state of affairs. We can make further corrections as they're made in the code?